docs/architecture/in-memory-queue.md

```diff
@@ -1,21 +1,23 @@
-# In-memory timer queue
-This queue is similar to normal persisted timer queue, but it exists in memory only and never gets  
-persisted. It is created with generic `MemoryScheduledQueueFactory`, but currently serves only
-[speculative Workflow Task](./speculative-workflow-task.md) timeouts, therefore the only queue this factory creates
-is `SpeculativeWorkflowTaskTimeoutQueue` which uses same task executor as normal timer queue:
-`TimerQueueActiveTaskExecutor`.
+# In-memory Timer Queue
 
-Implementation uses `PriorityQueue` by `VisibilityTimestamp`: a task on top is the task that
-executed next.
+This queue is similar to the normal persisted timer queue, but it exists only in memory, ie it 
+never gets persisted. It is created by a generic `MemoryScheduledQueueFactory`, but currently serves
+only [speculative Workflow Task](./speculative-workflow-task.md) timeouts. Therefore, the only queue
+this factory creates is `SpeculativeWorkflowTaskTimeoutQueue` which uses the same task executor as
+the normal timer queue: `TimerQueueActiveTaskExecutor`.
 
-In-memory queue supports only `WorkflowTaskTimeoutTask` and there are two timeout types
-enforced by in-memory queue: `SCHEDULED_TO_START` and `START_TO_CLOSE`.
+Its implementation uses a `PriorityQueue` sorted by `VisibilityTimestamp`: the task on top is the
+task that is executed next.
 
-Executor of `WorkflowTaskTimeoutTask` from in-memory queue is the same as for normal timer queue,
-although it does one extra check for speculative Workflow Task. It checks if a task being executed still the same
-as stored in mutable state (`CheckSpeculativeWorkflowTaskTimeoutTask`). This is because MS can lose and create
-a new speculative Workflow Task, which will be a different Workflow Task and a timeout task must be skipped for it. 
+The in-memory queue only supports `WorkflowTaskTimeoutTask` and it only enforces
+`SCHEDULED_TO_START` and `START_TO_CLOSE`.
```

Comment:
There are only two timeout types for any WFT. My intent here was to emphasize that only one type of timer tasks (`WorkflowTaskTimeoutTask`) is supported which enforce both timeouts. In your version, it sounds like normal WFT has other timeouts but for speculative we have only these 2.

Reply:
@alexshtin what's the other timeout type?

Reply:
For activity tasks there is also `SCHEDULE_TO_CLOSE` which is for all attempts. WFT doesn't have it, it is getting retried forever.

Reply:
Gotcha! Changed it to this now:

> The in-memory queue only supports `WorkflowTaskTimeoutTask`, and enforces the
`SCHEDULE_TO_START` and `START_TO_CLOSE` timeouts.